kedro-telemetry/kedro_telemetry/masking.py

```diff
@@ -72,22 +72,30 @@ def _get_cli_structure(
     return output
 
 
+def _mask_unrecognized_arg(arg: str, vocabulary: Set[str]) -> str:
+    return arg if arg in vocabulary else MASK
+
+
 def _mask_kedro_cli(cli_struct: Dict[str, Any], command_args: List[str]) -> List[str]:
     """Takes a dynamic vocabulary (based on `KedroCLI`) and returns
     a masked CLI input"""
     output = []
     vocabulary = _get_vocabulary(cli_struct)
```

Comment:
Potential opportunity for refactor: looks like `cli_struct` is only used to extract the vocabulary. Maybe `_mask_kedro_cli` should take `vocabulary` as a parameter instead?

Reply:
I agree, that's right, but let's remove that part in general according to your next comment.